src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -748,5 +748,14 @@ public static class Diagnostics {
 			defaultSeverity: DiagnosticSeverity.Error,
 			isEnabledByDefault: true
 		);
+
+		public static readonly DiagnosticDescriptor D2LPageDerivedMustBePartial = new DiagnosticDescriptor(
+			id: "D2L0101",
+			title: "Classes derived from D2LPage must be partial",
+			messageFormat: "Partial page objects can support dependency injection rather than having to rely on locators",
+			category: "Correctness",
+			defaultSeverity: DiagnosticSeverity.Info,

```

Comment:
Given you can land the version bump at the same time as the necessary edits to fix existing diagnostics, I think this would be fine as error?

Reply:
Back with VS2017(I think?) when I could manage to open Lms.sln, for diagnostics that we made codefixes for I'd apply it to the whole sln with VS. But otherwise yeah, I do rewrites separately. https://github.com/Brightspace/omsmith-scratch/tree/roslyn-refactoring is the tooling I use, but I think y'all have also put stuff together for yourselves these days?

Reply:
Think I have something working, so bringing this back to error severity